pandas/tests/groupby/test_frame_value_counts.py

```diff
@@ -308,66 +308,130 @@ def test_data_frame_value_counts_dropna(
     tm.assert_series_equal(result_frame_groupby, expected)
 
 
-@pytest.mark.parametrize("as_index", [False, True])
+@pytest.mark.parametrize("as_index", [True, False])
 @pytest.mark.parametrize(
     "observed, expected_index",
     [
         (
             False,
             [
-                ("FR", "male", "low"),
-                ("FR", "female", "high"),
-                ("FR", "male", "medium"),
-                ("FR", "female", "low"),
-                ("FR", "female", "medium"),
-                ("FR", "male", "high"),
-                ("US", "female", "high"),
-                ("US", "male", "low"),
-                ("US", "female", "low"),
-                ("US", "female", "medium"),
-                ("US", "male", "high"),
-                ("US", "male", "medium"),
+                ("FR", "high", "female"),
+                ("FR", "high", "male"),
+                ("FR", "low", "male"),
+                ("FR", "low", "female"),
+                ("FR", "medium", "male"),
+                ("FR", "medium", "female"),
+                ("US", "high", "female"),
+                ("US", "high", "male"),
+                ("US", "low", "male"),
+                ("US", "low", "female"),
+                ("US", "medium", "female"),
+                ("US", "medium", "male"),
```

Comment:
Why is this test changing values? In particular the order.

Reply:
I modified this test so it uses multiple groupers (`country` and `education` instead of `country` only), thus the second index level is the `education` column and is sorted by the groupby `sort=True` default behavior.

I will remove the `as_index` change.

Reply:
It looks like you've added the single grouper test back but with changes - this makes the diff very hard to follow. Would it be an issue to restore the original test fully?

Reply:
It just adds some slight redundancy with other single grouper tests. I restored the original test, now called `test_categorical_single_grouper_with_only_observed_categories`, with a small change to `observed=True` case regarding this fix

Reply:
@LucasG0 - there is too much changing here. Correct me if I'm wrong, but it appears the original test is not restored - it seems to be heavily modified.

Reply:
The original test on the main branch is called `test_categorical`, to be more explicit and to avoid confusion with other added tests, it is now called `test_categorical_single_grouper_with_only_observed_categories` ([link](https://github.com/pandas-dev/pandas/pull/46798/files#diff-94eb7e512970f337748151525e12830e7ca10c5e8acf4db15a00f56a9b2e6814R323)) and I added it in [this commit](https://github.com/pandas-dev/pandas/pull/46798/commits/fc8ea62187c89b8fd92aba071e54e378089edab4).

I slightly modified the original test in this PR to fix the expected behavior with `observed=True`. In the original test, the result would differ depending on `observed` while it should not - non grouping columns are not affected by `observed`, and grouping columns do not have non-observed values here, thus `observed` has no effect.